### PR TITLE
Update libs

### PR DESCRIPTION
**Pull request by ger-benjamin**

### CODE REVIEW

**Review by marionb — APPROVED** [review]
LGTM!
Tested on with the CANNA application.

**Review by marionb — CHANGES_REQUESTED** [review]
It looks to me that with these changes we are loosing the click/hover interactions in the plots.
Testing with the CANNA application I noted the following issues:
-  **scatter plot & line plot:** selecting a dot in the scatter plot is no longer possible
- **All plots:** hovering over data bar/point does nothing (it should open a pop up with extra information about the data plotted)
-  Interactions coming from the table looks working good.

**Comment by ger-benjamin** [comment]
That's true.

The problem come from here: https://github.com/d3/d3-selection/issues/189 (linked from https://github.com/d3/d3-selection/issues/191).

In short, that was not logical for the D3 team to give the parameters we received in the `on` (`onmouseover`) function. Now they send only the event and the data we hover.

As we iterate on axis data (like the x axis data), we receive the x data value in the `on` callback.

```
  // Iterate on data, not on axis data:

    const barGroups = this.chart.selectAll()
      // old way:  data(this.yData)
      .data(this.data) // new
      .enter()
      .append('g');

     // Use data directly, not index for one axis and data for the other (it's still possible but still strange)

    barGroups
      .append('rect')
      .attr('class', 'bar')
      .attr('x', 0)
      // old way.attr('y', d => this.yScale(d) - barHeight / 2)
      .attr('y', d => this.yScale(d[this.getYColumnName()]) - barHeight / 2)
      .attr('height', barHeight)
      // old way .attr('width', (d, i) => this.xScale(this.xData[i]))
      .attr('width', d => this.xScale(d[this.getXColumnName()]))
      .attr('fill', `rgb(${config.color.join(',')})`)
      .on('mouseover', this.handleMouseOverBars_.bind(this));

    // old: handleMouseOverBars_(yDatum: any, index: number, nodes: any): {
    handleMouseOverBars_(event, datum): {
     // old const element = nodes[index];
    const element = event.target;
    // old  const xValue = this.xData[index];
    const xValue = datum[this.getXColumnName()];
    const yValue = datum[this.getYColumnName()];

   // And if you want the "ǹodes" as before:
   const nodes = element.parent.parent;
   // That means you could get the index by finding the element in the list of nodes, but that's risky and ugly.
   }
```

But that was already a little bit strange to iterate on the x data and not on the whole data array.
Now, the code iterate on the whole data array. It looks more logical and it will works with the new `on` callback.

File: src/cartesian-chart.ts [hunk 1]
```diff
@@ -220,10 +230,31 @@ export class CartesianChart extends BaseD3ChartSVG {
     this.drawSVG();
   }
 
+  /**
+   * Returns the x axisColumn from the config.
```
Comment (marionb) [inline src/cartesian-chart.ts:234]:
```suggestion
   * Returns name of the x axisColumn from the config.
```

File: src/cartesian-chart.ts [hunk 2]
```diff
@@ -220,10 +230,31 @@ export class CartesianChart extends BaseD3ChartSVG {
...
+   */
+  getXColumnName(): string {
+    return this.config_?.xAxis?.axisColumn;
+  }
+
+  /**
+   * Returns the x axisColumn from the config.
```
Comment (marionb) [inline src/cartesian-chart.ts:241]:
```suggestion
   * Returns name of the y axisColumn from the config.
```

File: src/cartesian-chart.ts [hunk 3]
```diff
@@ -220,10 +230,31 @@ export class CartesianChart extends BaseD3ChartSVG {
...
+   */
+  getYColumnName(): string {
+    return this.config_?.yAxis?.axisColumn;
+  }
+
+  /**
+   * Returns the opposite y axisColumn from the config.
```
Comment (marionb) [inline src/cartesian-chart.ts:248]:
```suggestion
   * Returns name of the opposite y axisColumn from the config.
```

File: src/cartesian-chart.ts [hunk 4]
```diff
@@ -220,10 +230,31 @@ export class CartesianChart extends BaseD3ChartSVG {
...
+   */
+  getOppositeYColumnName(): string {
+    return this.config_?.oppositeYAxis?.axisColumn;
+  }
+
   /**
    * Returns the axisColumn after a check if data exist for this axis column key.
```
Comment (marionb) [inline src/cartesian-chart.ts:255]:
```suggestion
   * Returns the axisColumn name after a check if data exist for this axis column key.
```

**Review by marionb — COMMENTED** [review]
It looks good in general. Some minor notes.
I still need to test with the CANNA application

File: src/cartesian-chart.ts [hunk 5]
```diff
@@ -122,9 +122,7 @@ export class CartesianChart extends BaseD3ChartSVG {
   fontSizeForAxis: string;
   fontSizeForTitle: string;
 
-  xData: DataValue[];
-  yData: DataValue[];
-  oppositeYData: DataValue[];
+  data: DataRow[];
```
[the remark this answers is not in the document]
Comment (ger-benjamin) [reply src/cartesian-chart.ts:125]: Ok. I prefere `dataset` (without the uppercase S if it's okay to you. [Wikipedia](https://en.wikipedia.org/wiki/Data_set) says it's correct).

Then, why not rename this `dataRow`. This term is not really clear to me. Why not `datum` or `dataItem` ?
Again from Wikipedia:  _The data set lists values for each of the variables, such as height and weight of an object, for each member of the data set. Each value is known as a _**datum**_._

[the remark this answers is not in the document]
Comment (ger-benjamin) [reply src/cartesian-chart.ts:125]: And `dataValue` to ... `typedValue` ?

[the remark this answers is not in the document]
Comment (marionb) [reply src/cartesian-chart.ts:125]: - `dataset` is good for me
- `dataItem` I think is the better choice as it is more intuitive to understand. In any case it is a good idea to rename it

**Review by marionb — APPROVED** [review]
Except for the name changes and the minor change suggestions this looks good

File: src/cartesian-chart.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ger-benjamin) [reply src/cartesian-chart.ts]: Mhh at the end, I was not convinced to change this `DataValue`, I've left it.
Other changes are done

**Pull request merged**